optimus/io/jdbc.py

```diff
@@ -227,31 +244,35 @@ def execute(self, query, limit=None):
         """
 
         # play defensive with a select clause
-        if self.db_driver == "oracle":
-            alias = " t"
-        elif self.db_driver == "presto":
-            alias = ""
+        if self.db_driver == DriverResolver.ORACLE.__str__():
+            query = "(" + query + ") t"
+        elif self.db_driver == DriverResolver.PRESTO.__str__():
+            query = "(" + query + ")"
+        elif self.db_driver == DriverResolver.CASSANDRA.__str__():
+            query = query
         else:
-            alias = " AS t"
-
-        query = "(" + query + self._limit(limit) + ")" + alias
+            query = "(" + query + ") AS t"
 
         logger.print(query)
         logger.print(self.url)
 
         conf = Spark.instance.spark.read \
-            .format("jdbc") \
+            .format("jdbc" if not DriverResolver.CASSANDRA.__str__() else "org.apache.spark.sql.cassandra") \
             .option("url", self.url) \
-            .option("dbtable", query) \
```

Comment:
@eschizoid Any reason for not supplying the query here? I can't see the query specified in your code. `execute` method receives arbitrary queries so this might work for loading the complete data (or with some limit) it won't work with  arbitrary queries.

Reply:
@arpit1997 Good catch, this was removed by mistake will add it back

Reply:
Good catch, this was removed by mistake. Will add it back

Reply:
Fixed